includes/derivatives.inc

```diff
@@ -118,9 +118,33 @@ function islandora_ocr_get_uploaded_file(AbstractDatastream $datastream) {
  *   The options for tesseract.
  *
  * @return bool
- *   TRUE on success, FALSE otherwise.
+ *   Returns TRUE on success and FALSE on failure.
  */
 function islandora_ocr_derive_datastreams(AbstractDatastream $datastream, array $options) {
+  $derivatives = islandora_ocr_generate_datastreams($datastream, $options);
+  $ret = $derivatives['OCR'];
+  $ret = $ret && islandora_ocr_update_datastream($datastream->parent, $derivatives['OCR'], 'OCR');
+  $ret = $ret && $derivatives['HOCR'];
+  $ret = $ret && islandora_ocr_update_datastream($datastream->parent, $derivatives['HOCR'], 'HOCR');
+
+  $files = array_filter(array($source_file, $ocr_file, $hocr_file));
+  array_map('file_unmanaged_delete', $files);
+  return $ret;
+}
+
+/**
+ * Generates OCR/HOCR datastreams and returns references to them.
```

Comment:
@qadan,

It should be clarified that this function is generating files, not datastreams.

Thanks!

----
William Panting
_Developer_
*[discoverygarden inc.](http://www.discoverygarden.ca) | Managing Digital Content*